include/zephyr/bluetooth/bluetooth.h

```diff
@@ -517,6 +525,60 @@ size_t bt_data_get_len(const struct bt_data data[], size_t data_count);
  */
 size_t bt_data_serialize(const struct bt_data *input, uint8_t *output);
 
+struct bt_le_local_features {
```

Comment:
I assume we want to extend this struct whenever the spec defines a new feature set. Eventually this struct can become very large. Is this something we want to do?

Reply:
It's a good question. Similar to the other `info` structs, it's only stack allocated so it wouldn't be taking up much memory, but might run into stack overflows in some cases I guess. 

An alternative would be to do similar to what I did for ISO where the internal state uses the public state to store information: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/conn_internal.h#L182-L183

Today we are still doing an allocation + copy, but we could in theory return a `const` pointer directly to that struct, which would effectively require no additional memory.